contracts/football_bets.py

```
@@ -40,37 +98,65 @@ def get_match_result() -> str:
+        # ✅ Schema validation
+        if not isinstance(result_json, dict):
+            raise Exception("Invalid match result format")
+
+        if "winner" not in result_json or "score" not in result_json:
+            raise Exception("Missing required fields in match result")
+
+        # ✅ Type validation
+        try:
+            winner = int(result_json["winner"])
+        except (ValueError, TypeError):
+            raise Exception("Invalid winner value")
```

Ruff (0.15.7)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)